frontend/src/serviceWorker.js

```diff
@@ -1,141 +1,66 @@
-                'New content is available and will be used when all ' +
-                  'tabs for this page are closed. See https://bit.ly/CRA-PWA.'
-              );
+self.addEventListener("activate", (event) =>
+  event.waitUntil(self.clients.claim())
+);
 
-              // Execute callback
-              if (config && config.onUpdate) {
-                config.onUpdate(registration);
-              }
-            } else {
-              // At this point, everything has been precached.
-              // It's the perfect time to display a
-              // "Content is cached for offline use." message.
-              console.log('Content is cached for offline use.');
+// Cache fonts
+workbox.routing.registerRoute(
+  ({ url }) =>
+    url.origin === "https://fonts.googleapis.com" ||
+    url.origin === "https://fonts.gstatic.com",
+  new workbox.strategies.StaleWhileRevalidate({
+    cacheName: "google-fonts",
+  })
+);
 
-              // Execute callback
-              if (config && config.onSuccess) {
-                config.onSuccess(registration);
-              }
-            }
-          }
-        };
-      };
-    })
-    .catch(error => {
-      console.error('Error during service worker registration:', error);
-    });
-}
+// Network only strategy for backend requests
+workbox.routing.registerRoute(
+  ({ url }) => url.pathname.startsWith("/api/"),
+  new workbox.strategies.NetworkOnly()
+);
 
-function checkValidServiceWorker(swUrl, config) {
-  // Check if the service worker can be found. If it can't reload the page.
-  fetch(swUrl, {
-    headers: { 'Service-Worker': 'script' },
-  })
-    .then(response => {
-      // Ensure service worker exists, and that we really are getting a JS file.
-      const contentType = response.headers.get('content-type');
-      if (
-        response.status === 404 ||
-        (contentType != null && contentType.indexOf('javascript') === -1)
-      ) {
-        // No service worker found. Probably a different app. Reload the page.
-        navigator.serviceWorker.ready.then(registration => {
-          registration.unregister().then(() => {
-            window.location.reload();
-          });
-        });
-      } else {
-        // Service worker found. Proceed as normal.
-        registerValidSW(swUrl, config);
-      }
-    })
-    .catch(() => {
-      console.log(
-        'No internet connection found. App is running in offline mode.'
-      );
-    });
-}
+// Use a stale-while-revalidate strategy for all other requests.
+workbox.routing.setDefaultHandler(
+  new workbox.strategies.StaleWhileRevalidate()
+);
 
-export function unregister() {
-  if ('serviceWorker' in navigator) {
-    navigator.serviceWorker.ready
-      .then(registration => {
-        registration.unregister();
-      })
-      .catch(error => {
-        console.error(error.message);
-      });
-  }
-}
+workbox.routing.setCatchHandler(({ event }) => {
+  console.log(event);
+  return caches.match(offlinePage, {
```

Comment:
I would put in a check here to see if `event.request.destination === 'document'`, and only serve the offline HTML page if it's that type of request. Otherwise, you could accidentally serve an HTML response when, for instance, a request for an image is made.

There's more info at https://developers.google.com/web/tools/workbox/guides/advanced-recipes#comprehensive_fallbacks